### ISSUE TITLE
subplot() should bump xref/yref axis references

### ISSUE THREAD

**Issue by mike-wise** [ropensci/plotly#1019]
It is pretty well described in this StackOverflow post, and in my answer to it there: [post](http://stackoverflow.com/questions/44077754/plotly-shape-positions-using-subplot). Do you need me to copy the information here as well? 

There seem to be a number of bugs exposed there, maybe related, maybe not.

**Comment by cpsievert** [comment]
Shapes are automatically scaled/repositioned if you're using *paper* coordinates, for example:

```r
rect <- list(
  type = "rect",
  xref = "paper", yref = "paper",
  x0 = 0.5, x1 = 0.6, 
  y0 = 0.5, y1 = 0.6
)
p1 <- layout(plot_ly(), shapes = rect)
subplot(p1, p1)
```

I'm not sure if it generally makes sense to something similar for non-paper coordinates.

**Comment by mike-wise** [comment]
That is a good tip to generate workarounds. So thanks for that.

But people like to annotate their graphs in their "domain" coordinates, not the least because they do not actually have access to the "paper" coordinates when they are creating the plot. And if there is a well-defined mapping always available to the plotly when generating the plot, why shouldn't plotly just do the coordinate transformation? Is there a downside?

**Comment by cpsievert** [comment]
Ok, after some more thought, I can see how this might be useful. In this example, for instance, the `xref`/`yref` should be "bumped" by `subplot()` (but only if missing, I think).

```r
library(plotly)
library(dplyr)

mtcars %>%
  group_by(vs) %>%
  do(
    p = plot_ly(., x = ~wt, y = ~mpg) %>% 
      layout(shapes = list(
        type = "rect", x0 = ~min(wt), x1 = ~max(wt), y0 = ~min(mpg), y1 = ~max(mpg)
      ))
  ) %>%
  subplot()
```

**Comment by cpsievert** [comment]
Closed via #1332
